tests/unit/runtime/runtime_initialization.cpp

```diff
@@ -0,0 +1,32 @@
+//  Copyright (c) 2018 Nikunj Gupta
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+// Simple test verifying runtime initialization sequence
+
+#define _GNU_SOURCE
```

Comment:
Why is this necessary?

Reply:
@hkaiser I added it to write portable code. It provides access to various nonstandard GNU/Linux extension functions. I have not made any use of it, but I added keeping in mind that I might need it in the final implementation.

Reply:
I don't think it to be a good idea to request from the user to define this in order for your system to work. Wouldn't there be a way to just `#include <hpx/hpx_main.hpp>` and be done with it?